rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,400 @@
+            to_asset: Asset,
+            timestamp: Timestamp,
+            seconds: int | None = None,
+    ) -> bool:
+        return not self.is_penalized()
+
+    @staticmethod
+    def _get_asset_identifier(asset: AssetWithOracles) -> dict[str, str]:
+        """
+        May raise:
+        - UnsupportedAsset
+        """
+        if asset.is_evm_token() is True:
+            asset = asset.resolve_to_evm_token()
+            if asset.chain_id == ChainID.POLYGON_POS:
+                network = 'polygon-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_ONE:
+                network = 'arb-mainnet'
+            elif asset.chain_id == ChainID.BINANCE:
+                network = 'bnb-mainnet'
+            elif asset.chain_id == ChainID.AVALANCHE:
+                network = 'avax-mainnet'
+            elif asset.chain_id == ChainID.ZKSYNC_ERA:
+                network = 'zksync-mainnet'
+            elif asset.chain_id == ChainID.ETHEREUM:
+                network = 'eth-mainnet'
+            elif asset.chain_id == ChainID.POLYGON_ZKEVM:
+                network = 'polygonzkevm-mainnet'
+            elif asset.chain_id == ChainID.SCROLL:
+                network = 'scroll-mainnet'
+            elif asset.chain_id == ChainID.GNOSIS:
+                network = 'gnosis-mainnet'
+            elif asset.chain_id == ChainID.BASE:
+                network = 'base-mainnet'
+            elif asset.chain_id == ChainID.FANTOM:
+                network = 'fantom-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_NOVA:
+                network = 'arbnova-mainnet'
+            else:
+                raise UnsupportedAsset(asset.identifier)
+
+            return {'network': network, 'address': asset.evm_address}
+
+        # Alchemy team recommends using standard ticker symbols (e.g. ETH)
+        # over coingecko ids (e.g. ethereum) for better reliability.
+        return {'symbol': asset.symbol}
+
+    def _deserialize_current_price(
+            self,
+            result: list[dict[str, Any]],
+            asset_identifier: dict[str, str],
+            from_asset: Asset,
+            to_asset: Asset,
+    ) -> Price:
+        """Deserializes current price data from Alchemy API response.
+
+        If price data is missing, invalid, or there's an error, ZERO_PRICE is returned.
+        Otherwise, the deserialized price is returned.
+        """
+        try:
+            for entry in result:
+                if self._is_identifier_match(entry, asset_identifier) is False:
+                    continue
+
+                if 'error' in entry:
+                    log.warning(
+                        f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}. '  # noqa: E501
+                        f'API returned error: {entry["error"]}',
+                    )
+                    return ZERO_PRICE
+
+                if not (prices := entry['prices']):
```

Comment:
So this is a null check? Then here be explicit. Or can it be either `''` or `null` ?

```suggestion
                if not (prices := entry['prices']):
```

Reply:
no, it's to check if the array is empty